pkg/systemd/hwinfo.jsx

```diff
@@ -75,16 +75,62 @@ class HardwareInfo extends React.Component {
 
     render() {
         let pci = null;
+        let memory = null;
+        let $ = require("jquery");
 
         if (this.props.info.pci.length > 0) {
             let sortedPci = this.props.info.pci.concat();
             sortedPci.sort((a, b) => a[this.state.sortBy].localeCompare(b[this.state.sortBy]));
 
             pci = (
-                <Listing title={ _("PCI") } columnTitles={ [ _("Class"), _("Model"), _("Vendor"), _("Slot") ] }
-                         columnTitleClick={ index => this.setState({ sortBy: this.sortColumnFields[index] }) } >
-                    { sortedPci.map(dev => <ListingRow key={dev.slot} columns={[ dev.cls, dev.model, dev.vendor, dev.slot ]} />) }
-                </Listing>
+                <div id="pci_table">
+                    <Listing title={ _("PCI") } columnTitles={ [ _("Class"), _("Model"), _("Vendor"), _("Slot") ] }
+                             columnTitleClick={ index => this.setState({ sortBy: this.sortColumnFields[index] }) } >
+                        { sortedPci.map(dev => <ListingRow columns={[ dev.cls, dev.model, dev.vendor, dev.slot ]} />) }
+                    </Listing>
+                </div>
+            );
+        }
+
+        if (this.props.info.memory.array.length > 0) {
+            let empty_span = null;
+            let display_all = function(e) {
```

Comment:
assigning a function to a let is rather strange. Why not simply declare the function? `function display_all(e) {`